tests/typechecker/destructure_missing_variable.jakt

```diff
@@ -0,0 +1,8 @@
+/// Expect:
+/// - error: "Variable 'bar' not found"
+
+function main() {
+    mut foo = 0
+
+    (foo, bar) = (1, 2)
```

Comment:
Same here, the destructuring should go into bindings not assignments. `foo` already exists so unless you shadow it, you can't rebind it.